src/history/hs_rec.c

```diff
@@ -261,6 +261,33 @@ __hs_next_upd_full_value(WT_SESSION_IMPL *session, WT_UPDATE_VECTOR *updates,
     return (0);
 }
 
+/*
+ * __hs_failpoint --
+ *     A generic failpoint function, it will return EBUSY if the failpoint triggers. Takes a double
+ *     representing the probability of the failpoint occurring. Supports percentages with two
+ *     decimal places.
+ */
+static inline int
+__hs_failpoint(WT_SESSION_IMPL *session, double probability)
```

Comment:
We could make this function more generic by passing in the timing stress flag we're interested in but I don't think that's required for this ticket.